File: src/main/java/dev/sky_lock/pocketlifevehicle/vehicle/FourSeatsCar.java
package dev.sky_lock.pocketlifevehicle.vehicle;

import dev.sky_lock.pocketlifevehicle.vehicle.model.Model;
import org.bukkit.Location;

/**
 * @author sky_lock
 */

class FourSeatsCar extends Car {

    FourSeatsCar(Model model) {
        super(model);
    }

    @Override
    void spawn(Location location) {
        super.spawn(location);
        SeatArmorStand driver = new SeatArmorStand(getCenter().getWorld(), location.getX(), location.getY(), location.getZ());
        driver.assemble(this, SeatPosition.FOUR_DRIVER);
        SeatArmorStand passenger = new SeatArmorStand(getCenter().getWorld(), location.getX(), location.getY(), location.getZ());
        passenger.assemble(this, SeatPosition.FOUR_PASSENGER);
        SeatArmorStand rearRight = new SeatArmorStand(getCenter().getWorld(), location.getX(), location.getY(), location.getZ());
        rearRight.assemble(this, SeatPosition.FOUR_REAR_RIGHT);
        SeatArmorStand rearLeft = new SeatArmorStand(getCenter().getWorld(), location.getX(), location.getY(), location.getZ());
        rearLeft.assemble(this, SeatPosition.FOUR_REAR_LEFT);
        addSeat(driver);
        addSeat(passenger);
        addSeat(rearRight);
        addSeat(rearLeft);
        super.seats.forEach(getCenter().getWorld()::addEntity);
    }
}

File: src/main/kotlin/dev/sky_lock/pocketlifevehicle/vehicle/FourSeatsCar.kt
package dev.sky_lock.pocketlifevehicle.vehicle

import dev.sky_lock.pocketlifevehicle.vehicle.model.Model
import org.bukkit.Location
import java.util.function.Consumer

/**
 * @author sky_lock
 */
internal class FourSeatsCar(model: Model?) : Car(model!!) {
    override fun spawn(location: Location) {
        super.spawn(location)
        val driver = SeatArmorStand(center!!.getWorld(), location.x, location.y, location.z)
        driver.assemble(this, SeatPosition.FOUR_DRIVER)
        val passenger = SeatArmorStand(center!!.getWorld(), location.x, location.y, location.z)
        passenger.assemble(this, SeatPosition.FOUR_PASSENGER)
        val rearRight = SeatArmorStand(center!!.getWorld(), location.x, location.y, location.z)
        rearRight.assemble(this, SeatPosition.FOUR_REAR_RIGHT)
        val rearLeft = SeatArmorStand(center!!.getWorld(), location.x, location.y, location.z)
        rearLeft.assemble(this, SeatPosition.FOUR_REAR_LEFT)
        addSeat(driver)
        addSeat(passenger)
        addSeat(rearRight)
        addSeat(rearLeft)
        super.seats.forEach(Consumer { entity: SeatArmorStand? -> center!!.getWorld().addEntity(entity) })
    }
}